server/controllers/notificationController.js

```diff
@@ -1,17 +1,106 @@
+   * @param {object} req The notification's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the accomodation.
+   */
+  static async viewNotifications(req, res) {
+    const foundNotice = await Notification.findAll({
+      where: {
+        receiverId: req.user.id
+      }
+    });
+    if (foundNotice.length === 0) {
+      return res.status(404).json({
+        status: 404,
+        error: 'You dont have any notifications'
+      });
+    }
+    return res.status(200).json({
+      status: 200,
+      data: foundNotice
+    });
+  }
+
   /**
-     * This method handle the html file.
-     * @param {object} req The user's request.
-     * @param {object} res The user's response.
-     * @returns {object} The status and message.
-     * */
-  static async userNotification(req, res) {
-    return res.sendFile(`/notification.html`, { root: 'UI/html' });
+   * This method handle the accommodation request.
+   * @param {object} req The notification's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the accomodation.
+   */
+  static async markOne(req, res) {
+    const foundOneNotice = await Notification.findOne({
+      where: {
+        receiverId: req.user.id,
+        id: req.params.id
+      }
+    });
+    if (!foundOneNotice) {
+      return res.status(404).json({
+        status: 404,
+        error: 'Notification not found'
+      });
+    }
+    if (req.body.viewed === 'read') {
+      await Notification.update({ viewed: 'read' }, { where: { id: req.params.id } });
+      return res.status(200).json({
+        status: 200,
+        message: 'Notification marked as read'
+      });
+    }
+    if (req.body.viewed === 'unread') {
+      await Notification.update({ viewed: 'unread' }, { where: { id: req.params.id } });
+      return res.status(200).json({
+        status: 200,
+        message: 'Notification marked as unread'
+      });
+    }
+  }
+
+  /**
+   * This method handle the accommodation request.
+   * @param {object} req The notification's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the accomodation.
+   */
+  static async markAll(req, res) {
+    const foundManyNotice = await Notification.findOne({
+      where: {
+        receiverId: req.user.id
+      }
+    });
+    if (!foundManyNotice) {
+      return res.status(404).json({
+        status: 404,
+        error: 'Notifications not found'
+      });
+    }
+    if (req.body.viewed === 'read') {
+      await Notification.update({ viewed: 'read' }, { where: { receiverId: req.user.id } });
+      return res.status(200).json({
+        status: 200,
+        message: 'Notifications marked as read'
+      });
+    }
+    await Notification.update({ viewed: 'unread' }, { where: { receiverId: req.user.id } });
+    return res.status(200).json({
+      status: 200,
+      message: 'Notifications marked as unread'
```

Comment:
@manzi-guev you are calling too many db APIs. why finding and updating? can't update() works alon without separating find and Update.

Reply:
I first find out if a user has notifications that are assigned to him